docs/navigating-compliance-custom.md

```diff
@@ -0,0 +1,140 @@
+# Navigating Complance With Custom Logic
+
+## Caveats
+
+- The compliance `rule` must be unique for the Custom `config-type`.
+- The data provided can come from either setting via the API like JSON or via match_config like CLI. It is up to the operator to enforce.
+- Does not may any accommodations for adding to git.
+- Mixing/Matching string (or CLI type) and JSON type compliance rules is **NOT** supported. A device should only have compliance rules from one or the other, and it is up to the operator to enforce.
+  - Applying a `match_config` presumes it is CLI type and not having one presumes it is JSON type.
+- If the developer of the `get_custom_compliance` is not cognizant, the UI experience of the details may not always be obvious what the issues are.
+  - As an example, if the developer simply returns a `True` or `False` into the missing or extra dictionary, it will not be obvious to the user.
+- The developer is responsible for ensuring the proper data structure is maintained for the given rule.
+
+## Justification
+
+While the maintainers believe that the proper way to provide configuration compliance is the opinionated solutions provided, which compares intended
+state vs actual state, we are conscious that this may not always a viable solution for all organizations. As an example, complicated configurations may
```

Comment:
```suggestion
state vs actual state, we are conscious that this may not always be a viable solution for all organizations. As an example, complicated configurations may
```